packages/palette/src/elements/Swiper/progress.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Calculate the percentage of the rail that has been scrolled.
+ * If the left edge is at the total minus the viewport we are done scrolling.
+ * @param viewport window width
+ * @param total total width of the rail
+ * @param left left edge of scroll
+ */
+export const percentage = ({
+  viewport,
+  total,
+  left,
+}: {
+  viewport: number
+  total: number
+  left: number
+}) => {
+  return Math.floor((left / (total - viewport)) * 100)
+}
+
+/**
+ * Convert the progress into the active index, given an array of `length`.
+ * @param progress number from 1 to 100
+ * @param length length of an array
+ */
+export const index = ({
```

Comment:
(nonblocking) Maybe `index` could be named something a bit more descriptive? Its easy to see `index` and get it confused with old web baggage / conventions.